From c750f915a39a65a981f57d62a58ac3e74444323b Mon Sep 17 00:00:00 2001
From: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
Date: Thu, 11 Jan 2024 20:42:32 +0530
Subject: [PATCH] wifi:ath12k: fix 6GHz hidden ssid scan issue

Currently when use hidden ssid in 6 GHz,
the station vap is not bringup due to scan does not fatch the hidden ssid.

The broadcast probe request sent by FW during the scan could have
been closer to the end of dwell time (30msec). Channel switch time
 in WKK is 30 msec and so the channel switch might will impact
the reception of the ucast probe response from AP.

The STA itself would have moved to the next channel and
hence did not ACK the ucast probe response.
So the FW root cause of the problem is low dwell time causing
this Ucast probe response not ACKed and hence suggestion to host
is to increase the dwell time to 70msec for all channels on the scan

Signed-off-by: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/wmi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/wmi.c b/drivers/net/wireless/ath/ath12k/wmi.c
index 4765d63..1642505 100644
--- a/drivers/net/wireless/ath/ath12k/wmi.c
+++ b/drivers/net/wireless/ath/ath12k/wmi.c
@@ -2970,8 +2970,8 @@ void ath12k_wmi_start_scan_init(struct ath12k *ar,
 	arg->dwell_time_active = 50;
 	arg->dwell_time_active_2g = 0;
 	arg->dwell_time_passive = 150;
-	arg->dwell_time_active_6g = 40;
-	arg->dwell_time_passive_6g = 30;
+	arg->dwell_time_active_6g = 70;
+	arg->dwell_time_passive_6g = 70;
 	arg->min_rest_time = 50;
 	arg->max_rest_time = 500;
 	arg->repeat_probe_time = 0;
-- 
2.17.1

